From felipe.balbi@nokia.com  Fri Apr 23 14:04:50 2010
From: Sergei Shtylyov <sshtylyov@ru.mvista.com>
Date: Thu, 25 Mar 2010 13:14:28 +0200
Subject: musb_core: don't touch 'musb->clock' in musb_free()
To: Greg KH <greg@kroah.com>
Message-ID: <1269515673-27980-7-git-send-email-felipe.balbi@nokia.com>


From: Sergei Shtylyov <sshtylyov@ru.mvista.com>

Remove duplicate/unbalanced calls to clk_disable()/clk_put() in musb_free():

- clk_disable() is called by musb_platform_exit() just prior to this call;
- clk_put() is called by the callers of musb_free() prior to calling it...

Signed-off-by: Sergei Shtylyov <sshtylyov@ru.mvista.com>
Signed-off-by: Felipe Balbi <felipe.balbi@nokia.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/musb/musb_core.c |    5 -----
 1 file changed, 5 deletions(-)

--- a/drivers/usb/musb/musb_core.c
+++ b/drivers/usb/musb/musb_core.c
@@ -1857,11 +1857,6 @@ static void musb_free(struct musb *musb)
 	musb_platform_exit(musb);
 	musb_writeb(musb->mregs, MUSB_DEVCTL, 0);
 
-	if (musb->clock) {
-		clk_disable(musb->clock);
-		clk_put(musb->clock);
-	}
-
 #ifdef CONFIG_USB_MUSB_HDRC_HCD
 	usb_put_hcd(musb_to_hcd(musb));
 #else
